src/CachePlugin.php

```diff
@@ -36,11 +42,13 @@ class CachePlugin implements Plugin
 
     /**
      * @param CacheItemPoolInterface $pool
+     * @param StreamFactory          $streamFactory
      * @param array                  $options
      */
-    public function __construct(CacheItemPoolInterface $pool, array $options = [])
+    public function __construct(CacheItemPoolInterface $pool, StreamFactory $streamFactory, array $options = [])
     {
         $this->pool = $pool;
+        $this->streamFactory = $streamFactory;
         $this->defaultTtl = isset($options['default_ttl']) ? $options['default_ttl'] : null;
```

Comment:
I know this is not a current change, but shouldn't we use options resolver here?